### PR TITLE
[nova] add support for sentry DSN from secret

### PR DESCRIPTION
**Pull request by ruvr**
@fwiesel This could save you maintaining 4 fresh nova sentry DSN's..
just drop the former sentry_dsn from values.yaml in the regions where sentry has been re-installed..

### CODE REVIEW

**Comment by fwiesel** [comment]
Thanks, that is great.

@joker-official you wanted to change it. Is that one okay for you?

File: openstack/nova/templates/hypervisors/hypervisors-vmware-deployment-vct.yaml [hunk 1]
```diff
@@ -103,8 +108,17 @@ template: |
           command: ["dumb-init"]
           args: ["neutron-dvs-agent", "--config-file", "/etc/neutron/neutron.conf", "--config-file", "/etc/neutron/plugins/ml2/ml2-conf.ini", "--config-file", "/etc/neutron/plugins/ml2/ml2-vmware.ini"]
           env:
-          - name: SENTRY_DSN
-            value: {{ .Values.neutron.sentry_dsn | quote }}
+            {{- if .Values.sentry.enabled }}
```
Comment (joker-official) [inline openstack/nova/templates/hypervisors/hypervisors-vmware-deployment-vct.yaml]: @fwiesel If you check for capabilities (which nobody else does btw.), you'd have to check here, too.

**Comment by fwiesel** [comment]
Something like this?
I've added also a check on the existence of the sentry-operator.

**Comment by joker-official** [comment]
Is there a way to update the `.Values.sentry.enabled` with the capability-check? Looks quite ugly when checked everywhere.

**Comment by fwiesel** [comment]
To my knowledge, you can't compute values. The only way you can compute something is in a template.

If you want to go a minimalistic way, I would say there are two ways to go about it:

- Remove the check, and people should know that there is an order of deploying things (e.g. first the operator, then the app), and if they don't they get an error on deploying.
- Remove the option to enable/disable sentry for apps, and only check for the existence of the operator.

File: openstack/nova/templates/hypervisors/hypervisors-vmware-deployment-vct.yaml (hunk 1 above)
Comment (fwiesel) [reply openstack/nova/templates/hypervisors/hypervisors-vmware-deployment-vct.yaml]: Ah, I was planning to remove that part (See #471)

Comment (fwiesel) [reply openstack/nova/templates/hypervisors/hypervisors-vmware-deployment-vct.yaml]: Exactly for things like this, I think it is sensible to move the dvs agent out of the nova chart.
We need to copy values around and keep them in sync.

**Comment by fwiesel** [comment]
Personally, I am not a fan of having two places to configure, and you have to match them that it works. But here you.

**Comment by joker-official** [comment]
I get that, but my rationale is: If you look into the config and it's set to enabled, then it should be enabled and not only enabled if the sentry operator was deployed before or not. The place you'll look at are the values. You're not gonna read the chart for that. The deployment should rather fail and the user should sort it out.

**Comment by joker-official** [comment]
superseded by https://github.com/sapcc/helm-charts/pull/2173

**Pull request closed without merging**